app/models/git/repository.rb

```
@@ -34,11 +34,21 @@ def head_sha
     end
 
     def read_json_blob(commit, path)
-      oid = find_file_oid(commit, path)
-      raw = read_blob(oid, "{}")
+      raw = read_file_blob(commit, path, "{}")
       JSON.parse(raw, symbolize_names: true)
     end
 
+    def read_text_blob(commit, path)
+      read_file_blob(commit, path, "")
+    end
+
+    def read_file_blob(commit, path, default = nil)
+      oid = find_file_oid(commit, path)
+      read_blob(oid, default)
+    rescue Rugged::TreeError
+      default
+    end
```

This currently swallows the error, which might not be what we want. An option would be to allow passing a parameter that could indicate if the error should be swallowed or not, or something like that. A similar strategy is implemented in the `lookup_commit` method.

---

What's the situation where we don't want this?

---

I don't yet have an example. I thought that maybe you had an example, but if not, we can probably ignore this.